### PR TITLE
[Fluent 2 iOS] DateTimePicker + CalendarView update

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

`DateTimePicker` and `CalendarView` have been updated to match their fluent 2 design. 
The navigation bar for `DateTimePicker` is being set by `FluentUIFramework`. As of now, I have only added functionality to handle the custom nav bar specs for `DateTimePicker`. The entire class will be updated separately to use tokens instead of `Colors` in a separate PR.

### Verification

All the changes were tested on the demo app.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before_calendar_light | after_calendar_light |
| before_calendar_dark | after_calendar_dark |
| before_calendar_segmented_light | after_calendar_segmented_light |
| before_calendar_segmented_dark | after_calendar_segmented_dark |
| before_calendar_subtitle_light | after_calendar_subtitle_light |
| before_calendar_subtitle_dark | after_calendar_subtitle_dark |
| before_date_time_light | after_date_time_light |
| before_date_time_dark | after_date_time_dark |
| before_date_time_segmented_light | after_date_time_segmented_light |
| before_date_time_segmented_dark | after_date_time_segmented_dark |
| before_light | after_light |
| before_dark | after_dark |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1246)

### CODE REVIEW

**Comment by harrieshin** [comment]
i understand you might be following the design spec, but it seems bit weird the the previous dates are darker and future dates are lighter?  shouldn't selectable date be darker?

**Comment by harrieshin** [comment]
please update readme doc on datetimepicker as well

**Comment by harrieshin** [comment]
because there is font change, can you do a quick larger text test?

**Comment by laminesm** [comment]
> i understand you might be following the design spec, but it seems bit weird the the previous dates are darker and future dates are lighter? shouldn't selectable date be darker?

I agree, it seemed a bit odd but I just defaulted to implementing what design spec'ed. I'll reach out and see what they think.

File: ios/FluentUI/Calendar/Views/CalendarViewDayCell.swift [hunk 1]
```diff
@@ -187,21 +198,16 @@ class CalendarViewDayCell: UICollectionViewCell {
...
         switch textStyle {
         case .primary:
-            dateLabel.textColor = Colors.Calendar.Day.textPrimary
+            dateLabel.textColor = UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.foreground3])
         case .secondary:
-            dateLabel.textColor = Colors.Calendar.Day.textSecondary
+            dateLabel.textColor = UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.foreground1])
         }
 
-        switch backgroundStyle {
```
Comment (harrieshin) [inline ios/FluentUI/Calendar/Views/CalendarViewDayCell.swift:195]: with this removal, is there any usage of CalendarViewDayCellBackgroundStyle ?

File: ios/FluentUI/Core/FluentUIFramework.swift [hunk 2]
```diff
@@ -61,6 +61,20 @@ public class FluentUIFramework: NSObject {
...
 
+    enum NavigationBarStyle {
+        case normal
+        case dateTimePicker
+
+        func backgroundColor(fluentTheme: FluentTheme) -> UIColor {
+            switch self {
+            case .normal:
+                return UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.background3])
+            case.dateTimePicker:
```
Comment (harrieshin) [inline ios/FluentUI/Core/FluentUIFramework.swift:72]: nit: space after "case"

File: ios/FluentUI/Calendar/Views/CalendarViewDayMonthYearCell.swift [hunk 3]
```diff
@@ -47,16 +52,20 @@ class CalendarViewDayMonthYearCell: CalendarViewDayMonthCell {
...
-            yearLabel.textColor = Colors.Calendar.Day.textPrimary
+            yearLabel.textColor = UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.foreground3])
         case .secondary:
-            yearLabel.textColor = Colors.Calendar.Day.textSecondary
+            yearLabel.textColor = UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.foreground1])
         }
+    }
+
+    // Only supports indicator levels from 0...4
+    func setup(textStyle: CalendarViewDayCellTextStyle, backgroundStyle: CalendarViewDayCellBackgroundStyle, selectionStyle: CalendarViewDayCellSelectionStyle, monthLabelText: String, dateLabelText: String, yearLabelText: String, indicatorLevel: Int) {
```
Comment (sophialee0416) [inline ios/FluentUI/Calendar/Views/CalendarViewDayMonthYearCell.swift:65]: Nit: break down into paragraphs for readability

**Comment by laminesm** [comment]
> because there is font change, can you do a quick larger text test?

The contrast ratio issue found is the same as the issue with the activity indicator. Design suggested we ignore this issue until the FG colors are revisited. I have created a design ticket to track the controls currently affected.

File: ios/FluentUI/Calendar/Views/CalendarViewDayCell.swift (hunk 1 above)
Comment (laminesm) [reply ios/FluentUI/Calendar/Views/CalendarViewDayCell.swift:195]: There's no needed usage for it anymore. I'll ahead and remove it.

**Review by harrieshin — APPROVED** [review]

**Pull request merged**